tests/integration/index.ts

```diff
@@ -242,3 +242,308 @@ describe("Positive Cases", () => {
+
+      vBNXPrice = Number(await PriceOracle.getUnderlyingPrice(vBNX.address)) / 10 ** 18;
+      vBSWPrice = Number(await PriceOracle.getUnderlyingPrice(vBSW.address)) / 10 ** 18;
+
+      //Approve more assets for liquidation
+      await BSW.connect(acc1Signer).faucet(bswBorrowAmount);
+      await BSW.connect(acc1Signer).approve(vBSW.address, bswBorrowAmount);
+    });
+    it("Should revert when repay amount on liquidation is not equal to borrow amount", async function () {
+      //Repay amount does not make borrower principal to zero
+      const repayAmount = bswBorrowAmount / 2;
+      const param = {
+        vTokenCollateral: vBNX.address,
+        vTokenBorrowed: vBSW.address,
+        repayAmount: repayAmount,
+      };
+      await expect(Comptroller.connect(acc1Signer).liquidateAccount(acc2, [param])).to.be.revertedWith(
+        "Nonzero borrow balance after liquidation",
+      );
+    });
+    it("Should success on liquidation when repayamount is equal to borrowing", async function () {
+      const param = {
+        vTokenCollateral: vBNX.address,
+        vTokenBorrowed: vBSW.address,
+        repayAmount: bswBorrowAmount,
+      };
+      result = await Comptroller.connect(acc1Signer).liquidateAccount(acc2, [param]);
+      //Liquidation Mantissa is set for 1
+      const seizeTokens = (bswBorrowAmount * vBSWPrice) / vBNXPrice;
+      await expect(result)
+        .to.emit(vBSW, "LiquidateBorrow")
+        .withArgs(acc1, acc2, bswBorrowAmount, vBNX.address, seizeTokens.toFixed(0));
+
+      const liquidatorBalance = await vBNX.connect(acc1Signer).balanceOf(acc1);
+      expect(liquidatorBalance).to.equal(seizeTokens.toFixed(0));
+    });
+  });
+  describe("Liquidation of user via Vtoken", () => {
+    let mintAmount = convertToUnit("1", 8);
+    let acc1Signer: Signer;
+    let acc2Signer: Signer;
+    let bswBorrowAmount = convertToUnit("1", 4);
+
+    beforeEach(async () => {
+      acc1Signer = await ethers.getSigner(acc1);
+      acc2Signer = await ethers.getSigner(acc2);
+
+      await BNX.connect(acc2Signer).faucet(mintAmount);
+      await BNX.connect(acc2Signer).approve(vBNX.address, mintAmount);
+
+      // Fund 2nd account
+      await BSW.connect(acc1Signer).faucet(mintAmount);
+      await BSW.connect(acc1Signer).approve(vBSW.address, mintAmount);
+
+      await expect(vBNX.connect(acc2Signer).mint(mintAmount))
+        .to.emit(vBNX, "Mint")
+        .withArgs(acc2, mintAmount, mintAmount);
+      //borrow
+      //Supply WBTC to market from 2nd account
+      await expect(vBSW.connect(acc1Signer).mint(mintAmount))
+        .to.emit(vBSW, "Mint")
+        .withArgs(acc1, mintAmount, mintAmount);
+      await expect(vBSW.connect(acc2Signer).borrow(bswBorrowAmount))
+        .to.emit(vBSW, "Borrow")
+        .withArgs(acc2, bswBorrowAmount, bswBorrowAmount, bswBorrowAmount);
+
+      vBNXPrice = Number(await PriceOracle.getUnderlyingPrice(vBNX.address)) / 10 ** 18;
```

Comment:
We could use BigNumber and a scaling function here for consistency

Reply:
BigNumber is fixed.
What does scaling function explicitly mean?
Removed all redundant calculation of price in code.

Reply:
We could write a helper function similar to `convertToUnit` that scales a number down to its decimal based on a provided exponent.